### DISCUSSION TITLE
When should we release `1.0.0`?

### DISCUSSION THREAD

**Discussion by fpgmaas** [osprey-oss/deptry#868] [General]
`deptry` has recently celebrated its [second anniversary](https://github.com/fpgmaas/deptry/releases/tag/0.0.1). Currently, we are still on major version zero. The [Semantic Versioning](https://semver.org/) guidelines say the following about major version zero:

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

This raises the question: when do we consider the public API stable enough to warrant a `1.0.0` release for `deptry`?

---

I would argue that we have left the initial development phase, and the public API is currently stable enough for us to proceed with a `1.0.0` release. In the past several months, there have been very few significant changes to the public API. 

The only drawback of a `1.0.0` release is that we no longer can make breaking changes to the public API without increasing the major version. However, since the project has a relatively high amount of [downloads per month](https://pypistats.org/packages/deptry), it may be better for our users anyway to introduce any future breaking changes with a bump in the major version.

Curious for the thoughts of others on this :)

P.S. or do we simply adopt [ZeroVer](https://0ver.org/)?

**Comment by mkniewallner** [comment]
On board with the idea! I think we reached a point where things are stable enough to warrant a 1.0 release.

Before we do so, it might be a good time to think about breaking changes we could introduce before 1.0, knowing that after that, we would need to be more careful about them (although we can always bump major if needed, avoiding to make too many breaking changes and thus major releases would be nice).

I don't have that much things on my mind, but one thing that has bugged me for some time is the usage of regexes to define [exclusions](https://deptry.com/usage/#exclude) for files/directories. I personally find that using globs is less verbose, while still allowing enough customisations for use cases users might have. I can create an issue for this specific thing later on.
